### ISSUE TITLE
Result of a query returns direct access to cache object - wanted data binding?

### ISSUE THREAD

**Issue by Lennard-Dietz** [apollographql/apollo-client#5117]
**Intended outcome:**
When editing an object, that was returned from an apollo query, wo don't change the values in the apollo cache.

**Actual outcome:**
The returned object is the cache object, editing it and changing its properties and values are directly reflected in the cache.

Is this the intended behaviour? And if yes, is it stated in the documentation?

**Versions**
System:
    OS: Linux 4.9 Debian GNU/Linux 9 (stretch) 9 (stretch)
  Binaries:
    Node: 10.15.3 - /usr/local/bin/node
    Yarn: 1.13.0 - /usr/local/bin/yarn
    npm: 6.4.1 - /usr/local/bin/npm
  npmPackages:
    apollo-angular: ^1.6.0 => 1.6.0 
    apollo-angular-link-http: ^1.8.0 => 1.8.0 
    apollo-cache-inmemory: ^1.6.2 => 1.6.2 
    apollo-client: ^2.6.3 => 2.6.3 
    apollo-link: ^1.2.12 => 1.2.12 
    apollo-link-context: ^1.0.18 => 1.0.18 
    apollo-link-error: ^1.1.11 => 1.1.11 
    apollo-server-express: ^2.6.7 => 2.7.0 
    graphql-codegen-typescript-apollo-angular: ^0.18.2 => 0.18.2

**Comment by chaunceyau** [comment]
Can you explain this a bit more?

**Comment by hwillson** [comment]
Please try a recent version of `@apollo/client` and let us know if this issue is still happening. Thanks!

**Comment by McPo** [comment]
@hwillson I have just came across this issue. I have updated vue-apollo to 3.0.4 but the issue remains.

Essentially if I bind the result of a apollo query to a input text field. Then further edit that text.
If the component was to be destroyed and recreated, my edits would still exist.

This contrast with how it would work if I was binding this directly to a `data` variable.
It appears that my edits are being saved to the cache without any mutation being actioned.

So if you go to edit a users name in a dialog box (pre-populated by a query with their current name).
Change it but then decide you don't wish to submit the change, so you hit cancel, if you were to reopen the dialog again your edits would still be there.

I have provided a codesandbox example of the issue https://codesandbox.io/s/apollovue-apollodata-cache-issue-5pq2z?file=/src/SimpleDialog.vue

Cheers,
Emmet
